crates/starknet-devnet-core/src/starknet/mod.rs

```diff
@@ -1182,15 +1197,20 @@ impl Starknet {
         let mut transactional_state =
             CachedState::new(CachedState::create_transactional(&mut state.state));
 
-        for (blockifier_transaction, transaction_type, skip_validate_due_to_impersonation) in
-            blockifier_transactions.into_iter()
+        for (tx_i, (blockifier_tx, transaction_type, skip_validate_due_to_impersonation)) in
+            blockifier_transactions.into_iter().enumerate()
         {
-            let tx_execution_info = blockifier_transaction.execute(
-                &mut transactional_state,
-                &block_context,
-                !skip_fee_charge,
-                !(skip_validate || skip_validate_due_to_impersonation),
-            )?;
+            let tx_execution_info = blockifier_tx
+                .execute(
+                    &mut transactional_state,
+                    &block_context,
+                    !skip_fee_charge,
+                    !(skip_validate || skip_validate_due_to_impersonation),
+                )
+                .map_err(|e| Error::ContractExecutionErrorInSimulation {
+                    failure_index: tx_i as u64,
+                    error_stack: gen_tx_execution_error_trace(&e),
+                })?;
```

Comment:
As per the specification of `starknet_simulateTransactions`, we shouldn't return the error right away, but include the error details in the trace of the result. Shall be taken care of in #628.